### PR TITLE
[SPARK-16430][SQL][STREAMING] Add option maxFilesPerTrigger

### PR DESCRIPTION
**Pull request by tdas**
## What changes were proposed in this pull request?

An option that limits the file stream source to read 1 file at a time enables rate limiting. It has the additional convenience that a static set of files can be used like a stream for testing as this will allows those files to be considered one at a time.

This PR adds option `maxFilesPerTrigger`.
## How was this patch tested?

New unit test

### CODE REVIEW

**Comment by tdas** [comment]
@marmbrus @zsxwing

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala [hunk 1]
```diff
@@ -26,6 +27,7 @@ import org.apache.spark.internal.Logging
 import org.apache.spark.sql.{DataFrame, Dataset, SparkSession}
 import org.apache.spark.sql.execution.datasources.{CaseInsensitiveMap, DataSource, ListingFileCatalog, LogicalRelation}
 import org.apache.spark.sql.types.StructType
+import org.apache.spark.util.Utils
```
Comment (marmbrus) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala]: Where is this used?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala [hunk 2]
```diff
@@ -45,6 +47,7 @@ class FileStreamSource(
   private val qualifiedBasePath = fs.makeQualified(new Path(path)) // can contains glob patterns
   private val metadataLog = new HDFSMetadataLog[Seq[String]](sparkSession, metadataPath)
   private var maxBatchId = metadataLog.getLatest().map(_._1).getOrElse(-1L)
+  private val maxFilesPerBatch = getMaxFilesPerBatch()
```
Comment (marmbrus) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala]: Maybe some scaladoc here about what this parameter does / its purpose.

**Comment by marmbrus** [comment]
LGTM

**Comment by tdas** [comment]
@marmbrus I have also added the option to the docs of the load methods in DataStreamReader.

**Comment by tdas** [comment]
Merging this to master and 2.0

**Pull request closed without merging**